com.unity.formats.fbx/Editor/ConvertToNestedPrefab.cs

```diff
@@ -207,6 +210,54 @@ internal static bool SetContainsOnlyPrefabAssets(Object[] toConvert)
             }
             return converted.ToArray();
         }
+
+        /// <summary>
+        /// For sceneObj, replace reference of origObj with newObj.
+        /// 
+        /// If the scene object is toConvertRoot or a child of it, then do not fix its references as it
+        /// will be deleted after conversion.
+        /// </summary>
+        /// <param name="sceneObj">scene object with reference to replace</param>
+        /// <param name="origObj">original object</param>
+        /// <param name="newObj">new object</param>
+        /// <param name="toConvertRoot">root of the hierarchy being converted</param>
+        internal static void FixSceneReferenceToObject(Object sceneObj, Object origObj, Object newObj, GameObject toConvertRoot)
+        {
+            // item has reference to origObj that need to be replaced by references to newObj
+            var go = ModelExporter.GetGameObject(sceneObj);
+            if (go && go.transform.IsChildOf(toConvertRoot.transform))
+            {
+                // if this is a child of what we are converting, don't update its references.
+                return;
+            }
+
+            var components = go.GetComponents<Component>();
+            foreach (var component in components)
+            {
+                var serializedComponent = new SerializedObject(component);
+                var property = serializedComponent.GetIterator();
+                property.Next(true); // skip generic field
+                // For SkinnedMeshRenderer, the bones array doesn't have visible children, but may have references that need to be fixed.
+                // For everything else, filtering by visible children in the while loop and then copying properties that don't have visible children,
+                // ensures that only the leaf properties are copied over. Copying other properties is not usually necessary and may break references that
+                // were not meant to be copied.
+                while (property.Next((component is SkinnedMeshRenderer) ? property.hasChildren : property.hasVisibleChildren))
+                {
+                    if (!property.hasVisibleChildren)
+                    {
+                        // with Undo operations, copying m_Father reference causes issues. Also, it is not required as the reference is fixed when
+                        // the transform is parented under the correct hierarchy (which happens before this).
+                        if (property.propertyType == SerializedPropertyType.ObjectReference && property.propertyPath != "m_GameObject" &&
+                            property.propertyPath != "m_Father" && property.objectReferenceValue &&
+                            (property.objectReferenceValue == origObj))
```

Comment:
this is really complicated. Can you do one condition per line?
```
if ( (property.PropertyType == SerializedPropertyType.ObjectReference)
  && (property.propertyPath != "m_GameObject")
  && (property.propertyPath != "m_Father")
...
```

Or rewrite it as:
```
if (property.propertyType != SerializedPropertyType.ObjectReference) continue;
if (property.propertyPath == "m_GameObject") continue;
...
```